pkg/controller/common/reconciler/secret.go

```diff
@@ -170,17 +172,41 @@ func GarbageCollectSoftOwnedSecrets(c k8s.Client, deletedOwner types.NamespacedN
 	return nil
 }
 
-// GarbageCollectAllSoftOwnedOrphanSecrets iterates over all Secrets that reference a soft owner. If the owner
+// GarbageCollectAllSoftOwnedOrphanSecrets iterates over all Secrets in the namespaces that the operator
+// manages that reference a soft owner. If the owner
 // doesn't exist anymore, it deletes the secrets.
 // Should be called on operator startup, after cache warm-up, to cover cases where
 // the operator is down when the owner is deleted.
 // If the operator is up, garbage collection is already handled by GarbageCollectSoftOwnedSecrets on owner deletion.
-func GarbageCollectAllSoftOwnedOrphanSecrets(c k8s.Client, ownerKinds map[string]client.Object) error {
+func GarbageCollectAllSoftOwnedOrphanSecrets(c k8s.Client, ownerKinds map[string]client.Object, managedNamespaces []string) error {
+	if len(managedNamespaces) == 0 {
```

Comment:
Would it be possible to get the same behavior by adding a predicate while iterating over the SecretList:

```suggestion
	// remove any secret whose owner in the same namespace doesn't exist
	for i := range secrets.Items {
		secret := secrets.Items[i]
		if !predicates.IsNamespaceManaged(secret.Namespace, managedNamespaces) {
			continue
		}
```

Reply:
I'm not following you here.  GarbageCollectAllSoftOwnedOrphanSecrets is called asynchronously a single time at startup, and doesn't have a `corev1.SecretList` at this time.  Are you referring to leveraging `predicates.IsNamespaceManaged` in this func?
https://github.com/elastic/cloud-on-k8s/blob/0dbe3b868cbee9cbcdb854ccae6ac688fdad6917/pkg/controller/common/reconciler/secret.go#L178

Reply:
Sorry, just realized that I was not clear.
Yes, I was wondering if we could keep the existing `GarbageCollectAllSoftOwnedOrphanSecrets` and reuse the predicate. I think it would avoid to introduce `garbageCollectSecrets` (no strong opinion though as it is only a cosmetic change).

Reply:
Ok, I finally understood what you were suggesting, and since I thought it was this call which was causing the namespace error

`if err := c.List(context.Background()`

upon doing a little more debugging, I noticed it was this:

`err := c.Get(context.Background(), types.NamespacedName{Namespace: softOwner.Namespace, Name: softOwner.Name}, owner)`

and it all made sense.  This greatly simplifies these changes in this package.  Thanks @barkbay